selfdrive/camerad/cameras/camera_webcam.cc

```diff
@@ -0,0 +1,245 @@
+#include "camera_webcam.h"
+
+#include <unistd.h>
+#include <string.h>
+#include <signal.h>
+#include <pthread.h>
+
+#include "common/util.h"
+#include "common/timing.h"
+#include "common/swaglog.h"
+#include "buffering.h"
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/core.hpp>
+#include <opencv2/videoio.hpp>
+
+extern volatile sig_atomic_t do_exit;
+
+#define FRAME_WIDTH  1164
+#define FRAME_HEIGHT 874
+
+namespace {
+void camera_open(CameraState *s, VisionBuf *camera_bufs, bool rear) {
+  assert(camera_bufs);
+  s->camera_bufs = camera_bufs;
+}
+
+void camera_close(CameraState *s) {
+  tbuffer_stop(&s->camera_tb);
+}
+
+void camera_release_buffer(void *cookie, int buf_idx) {
+  CameraState *s = static_cast<CameraState *>(cookie);
+}
+
+void camera_init(CameraState *s, int camera_id, unsigned int fps) {
+  assert(camera_id < ARRAYSIZE(cameras_supported));
+  s->ci = cameras_supported[camera_id];
+  assert(s->ci.frame_width != 0);
+
+  s->frame_size = s->ci.frame_height * s->ci.frame_stride;
+  s->fps = fps;
+
+  tbuffer_init2(&s->camera_tb, FRAME_BUF_COUNT, "frame", camera_release_buffer, s);
+}
+
+static void* rear_thread(void *arg) {
+  int err;
+
+  set_thread_name("webcam_rear_thread");
+  CameraState* s = (CameraState*)arg;
+
+  cv::VideoCapture cap_rear(0); // road
+  cap_rear.set(cv::CAP_PROP_FRAME_WIDTH, 1280);
+  cap_rear.set(cv::CAP_PROP_FRAME_HEIGHT, 720);
+  cap_rear.set(cv::CAP_PROP_FPS, s->fps);
+  cap_rear.set(cv::CAP_PROP_AUTOFOCUS, 0); // off
+  cap_rear.set(cv::CAP_PROP_FOCUS, 0); // 0 - 255?
+  cv::Rect roi_rear(160, 0, 960, 720);
+
+  if (!cap_rear.isOpened()) {
+    err = 1;
+  }
+
+  uint32_t frame_id = 0;
+  TBuffer* tb = &s->camera_tb;
+
+  while (!do_exit) {
+    cv::Mat frame_mat;
+
+    cap_rear >> frame_mat;
+
+    // int rows = frame_mat.rows;
+    // int cols = frame_mat.cols;
+    // printf("Raw Rear, R=%d, C=%d\n", rows, cols);
+
+    cv::Mat cropped_mat = frame_mat(roi_rear);
+    cv::resize(cropped_mat, frame_mat, cv::Size(s->ci.frame_width, s->ci.frame_height));
```

Comment:
Can you do the proper transform to eon intrinsics? https://github.com/commaai/openpilot/blob/5eeb8a0e02c69bdb1201bf60915a8de51ee74ef8/common/transformations/camera.py#L10